packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -41,77 +47,97 @@ function TemplateLabel( { template, homeId } ) {
 	onAddTemplateId,
+	onRemoveTemplateId,
 } ) {
 	const [ hoveredTemplate, setHoveredTemplate ] = useState();
 	const [ themePreviewVisible, setThemePreviewVisible ] = useState( false );
 
-	const onHoverTemplate = ( id ) => {
-		setHoveredTemplate( { id, type: 'template' } );
-	};
 	const onHoverTemplatePart = ( id ) => {
 		setHoveredTemplate( { id, type: 'template-part' } );
 	};
-
 	const onMouseEnterTheme = () => {
 		setThemePreviewVisible( () => true );
 	};
 	const onMouseLeaveTheme = () => {
 		setThemePreviewVisible( () => false );
 	};
 
-	const { currentTheme, templates, templateParts } = useSelect(
+	const { currentTheme, template, templateParts } = useSelect(
 		( select ) => {
 			const { getCurrentTheme, getEntityRecord } = select( 'core' );
+			const _template = getEntityRecord(
+				'postType',
+				'wp_template',
+				activeId
+			);
 			return {
 				currentTheme: getCurrentTheme(),
-				templates: ids.map( ( id ) => {
-					const template = getEntityRecord(
-						'postType',
-						'wp_template',
-						id
-					);
-					return {
-						label: template ? (
-							<TemplateLabel
-								template={ template }
-								homeId={ homeId }
-							/>
-						) : (
-							__( 'Loading…' )
-						),
-						value: id,
-						slug: template ? template.slug : __( 'Loading…' ),
-					};
-				} ),
+				template: {
+					label: _template ? (
+						<TemplateLabel
+							template={ _template }
+							homeId={ homeId }
+						/>
+					) : (
+						__( 'Loading…' )
+					),
+					value: activeId,
+					slug: _template ? _template.slug : __( 'Loading…' ),
+					content: _template?.content,
+				},
 				templateParts: templatePartIds.map( ( id ) => {
-					const template = getEntityRecord(
+					const templatePart = getEntityRecord(
 						'postType',
 						'wp_template_part',
 						id
 					);
 					return {
-						label: template ? (
-							<TemplateLabel template={ template } />
+						label: templatePart ? (
+							<TemplateLabel template={ templatePart } />
 						) : (
 							__( 'Loading…' )
 						),
 						value: id,
-						slug: template ? template.slug : __( 'Loading…' ),
+						slug: templatePart
+							? templatePart.slug
+							: __( 'Loading…' ),
 					};
 				} ),
 			};
 		},
-		[ ids, templatePartIds, homeId ]
+		[ activeId, templatePartIds, homeId ]
 	);
-	const [ isAddTemplateOpen, setIsAddTemplateOpen ] = useState( false );
+
+	const { saveEntityRecord } = useDispatch( 'core' );
+
+	const overwriteSlug =
+		TEMPLATE_OVERRIDES[ page.type ] &&
+		page.slug &&
+		TEMPLATE_OVERRIDES[ page.type ]( page.slug );
+	const overwriteTemplate = async () => {
+		const newTemplate = await saveEntityRecord( 'postType', 'wp_template', {
+			slug: overwriteSlug,
+			title: overwriteSlug,
+			status: 'publish',
+			content: template.content.raw,
+		} );
+		onAddTemplateId( newTemplate.id );
+	};
+	const unoverwriteTemplate = async () => {
+		await apiFetch( {
+			path: `/wp/v2/templates/${ template.value }`,
```

Comment:
Should the core/data controls handle this?

Reply:
Yeah